calico/openstack/mech_calico.py

```diff
@@ -607,6 +621,11 @@ def felix_heartbeat_thread(self, hostname):
                 else:
                     LOG.error("Unexpected response from Felix on %s: %s"
                               % (hostname, rsp))
+            except Again:
+                LOG.error("No response from Felix within allowed time (%sms)" %
+                          HEARTBEAT_RESPONSE_TIMEOUT)
```

Comment:
The comment not so many lines above says that you do raise the exception, and I think should be tidied up (presumably a bad block copy). Then I'm happy to merge.